node/primitives/src/disputes/mod.rs

```diff
@@ -66,6 +66,35 @@ impl CandidateVotes {
 }
 
 impl SignedDisputeStatement {
+	/// Create a new `SignedDisputeStatement` from information
+	/// that is available on-chain, and hence already can be trusted.
+	///
+	/// Attention: Not to be used other than with guaranteed fetches.
+	pub fn new_unchecked_from_trusted_source(
+		dispute_statement: DisputeStatement,
+		candidate_hash: CandidateHash,
+		session_index: SessionIndex,
+		validator_public: ValidatorId,
+		validator_signature: ValidatorSignature,
+	) -> Self {
+		assert_eq!(
```

Comment:
Codebase policy is to prove that panickers cannot fail. I think that `new_unchecked_from_trusted_source` should simply not do this check, or it should return an error.